myapp/app/views/tasks/index.html.erb

```diff
@@ -1,37 +1,58 @@
-<%= render "form", task: @new_task, btn_label: "Create" %>
+<div class="row">
+  <div class="col-2 bg-white rounded">
+    <%= form_with url: root_path, method: :get do |f| %>
+      <div class="mb-3 mt-3">
+        <div class="input-group">
+          <%= f.text_field :query, class: "form-control", placeholder: t('search_by_title'), value: params[:query] %>
+          <%= f.button raw('<i class="bi bi-search"></i>'), name: nil, class: "btn btn-light border-light-subtle", id: "btn-search" %>
+          <% unless params[:sort].to_s.empty? %>
+          <%= f.hidden_field :sort, value: params[:sort] %>
+          <% end %>
+        </div>
+      </div>
+      <div class="mb-3">
+      <%= f.select :status, options_for_select(Task.statuses.invert.map{ |k, v| [ I18n.t(v), k ] }, params[:status]), { include_blank: t('status_all') }, {class: "form-select", id: "search-status", onchange: "submit(this.form)" } %>
+      </div>
+    <% end %>
+  </div>
 
-<table class="table table-hover">
-  <thead>
-    <tr class="small">
-      <th scope="col"><%= t 'title' %></th>
-      <th scope="col"><%= t 'description' %></th>
-      <th scope="col">
-        <%= link_to_unless(params[:sort].eql?('created_at'), raw(t 'created_at' ) + raw('<i class="bi bi-sort-up"></i>'), root_path(:sort => 'created_at')) %>
-      </th>
-      <th scope="col">
-        <%= link_to_unless(params[:sort].eql?('due_date_at'), raw(t 'due_date_at' ) + raw('<i class="bi bi-sort-up"></i>'), root_path(:sort => 'due_date_at')) %>
-      </th>
-      <th scope="col"></th>
-    </tr>
-  </thead>
-<% if @tasks.any? %>
-  <tbody>
-<% @tasks.each do |task| %>
-    <tr>
-      <td><%= link_to task.title, task_path(task) %></td>
-      <td><%= task.description %></td>
-      <td><%= task.created_at %></td>
-      <td><%= task.due_date_at.to_s.split[0] %></td>
-      <td>
-      <%= button_to "Delete",
-      task_path(task),
-      method: :delete,
-      class: "btn btn-danger btn-xs del-btn",
-      data: { "turbo-method": :delete, "turbo-confirm": 'Are you sure?' }
-      %>
-      </td>
-    </tr>
-<% end %>
-  </tbody>
-<% end %>
-</table>
+  <div class="col-10">
+    <%= render "form", task: @new_task, btn_label: "Create" %>
+    <table class="table table-hover">
+      <thead>
+        <tr>
+          <th scope="col"><%= t "title" %></th>
+          <th scope="col"><%= t "status" %></th>
+          <th scope="col">
+          <%= link_to_unless(params[:sort].eql?('created_at'), raw(t 'created_at' ) + raw('<i class="bi bi-sort-up"></i>'), root_path(request.parameters.merge(sort: 'created_at'))) %>
+          </th>
+          <th scope="col">
+          <%= link_to_unless(params[:sort].eql?('due_date_at'), raw(t 'due_date_at' ) + raw('<i class="bi bi-sort-up"></i>'), root_path(sort: 'due_date_at'))) %>
```

Comment:
```suggestion
          <%= link_to_unless(params[:sort].eql?('due_date_at'), raw(t 'due_date_at' ) + raw('<i class="bi bi-sort-up"></i>'), root_path(request.parameters.merge(sort: 'due_date_at'))) %>
```
 you can delete build_request_uri method.

Reply:
Just deleted the func as well.

Reply:
I confirmed your changes. Also, please check my suggestion for the above.